### DISCUSSION TITLE
IOS muiltiple options  but only 1 in Browser

### DISCUSSION THREAD

**Discussion by thorrrr** [dani-garcia/vaultwarden#3979] [Q&A] [answered]
HI 
One question i have never been able to solve is the multiple long list of password options in IOS.
I have Bitwarden on my IOS devices and works a dream untill i login to my server(s) apps 36 of them. 
I have them set in BW as exact so it matches server ip and port. Now in Browsers it only shows 1 entry for each app, which what i would expect.
But if i go and look in IOS i get like 25 options  its not able to pick up on ip and port i do not think.
Is this a limitations created by Apple or do i need to change some settings.

https://i.imgur.com/su4ns4N.png

**Comment by stefan0xC** [comment]
Since it works in your browser I think you have not changed the [default match detection](https://bitwarden.com/help/uri-match-detection/#default-match-detection) in your IOS apps?

If you just want to match server IP and Port you should select `Host`. `Exact` is only if the URI matches exactly (i.e. also the scheme, path and query string must match).

**Reply by thorrrr** [reply]
Would i be better just to go through my apps on ios and change to host?
Or you saying global settings?

PS Follow-up if i may 
How do you set IOS VW to recognise apps so it can offer password ?

**Reply by stefan0xC** [reply]
If you set it globally in the app settings you don't have to set it on each item (unless you want a different behavior for a specific item).

**Reply by thorrrr** [reply]
HI
That has not worked 100% i still see the long list and everything is set to host!
It looks like it saved in IOS by VW how do i purge the saved passwords from VW in IOS ?
I have fully removed the BW app from IOS and check and there are no options given.
So i can say its BW offering the saved passwords on fully deleted the app and data.
Checked a website obviously no options given. Added BW back added self hosting and details back.
Changed to HOST only also all the server apps are set to host 
When i check again i get a full long list of 25 + apps the ones i changed to HOST !!!!!!!

Also from above 
PS Follow-up if i may
How do you set IOS VW to recognise apps so it can offer password ?

**Reply by BobWs** [reply]
Having the same problem for years, it has never worked properly for me on iOS devices. I have tried all the offered options but none of them seem to work (Always the long list).
In contrast to 1Password on iOS which does work properly recognizing subdomains and IP addresses.
I think it is a problem with the Bitwarden iOS app.

**Reply by stefan0xC** [reply]
That's weird. Not sure why this happens. You can [search the bitwarden community forum](https://community.bitwarden.com/) or [check the bitwarden/mobile bug tracker](https://github.com/bitwarden/mobile/issues) if someone has experienced this issue.

ps. Please read the page I've already linked because it has [a section called **Obtaining URIs for mobile apps**](https://bitwarden.com/help/uri-match-detection/#obtaining-uris-for-mobile-apps).

**Reply by thorrrr** [reply, marked as answer]
FYI 

![image](https://github.com/dani-garcia/vaultwarden/assets/8170673/365b6f4f-30c0-498b-925c-84cd83a9af64)


I might ask them to make their own keyboard as its an Apple rule based on their keyboard

**Comment by thorrrr** [comment]
I have added a suggestion [here](https://community.bitwarden.com/t/bitwarden-to-make-their-own-keyboard-for-ios/59151)
